drivers/interrupt_controller/intc_plic.c

```diff
@@ -39,7 +39,16 @@
  * However, it is defined and supported by at least the Andes & Telink datasheet, and supported
  * in Linux's SiFive PLIC driver
  */
+#define PLIC_DRV_HAS_COMPAT(compat)                                                                \
+	DT_NODE_HAS_COMPAT(DT_COMPAT_GET_ANY_STATUS_OKAY(DT_DRV_COMPAT), compat)
+
+#if PLIC_DRV_HAS_COMPAT(andestech_plic)
+#define PLIC_REG_TRIG_TYPE_OFFSET 0x1080
+#elif PLIC_DRV_HAS_COMPAT(telink_plic)
 #define PLIC_REG_TRIG_TYPE_OFFSET 0x1080
+#else
+#define PLIC_REG_TRIG_TYPE_OFFSET 0 /* Not supported */
```

Comment:
In the long run, it would still be ideal to get another rev of the plic spec, but design by committee can take time.

Reply:
Hi @ycsin ,

It's up to you, but why not to make this register configurable via CONFIG option, e.g. like it was implemented before:
https://github.com/zephyrproject-rtos/zephyr/commit/8db1a5add221b5e47ad4e01dfcb3b4064a8299f4

Then the unsupported functionality would be compiled-out entirely, you wouldn't need to have a reserved offset value with special meaning, you wouldn't need to alter intc_plic.c every time new vendor supports this, etc.

Reply:
> why not to make this register configurable via CONFIG option, e.g. like it was implemented before:
> [8db1a5a](https://github.com/zephyrproject-rtos/zephyr/commit/8db1a5add221b5e47ad4e01dfcb3b4064a8299f4)

I like to keep the `sifive,plic-1.0.0` to closely resembles that of Linux kernel. Kconfig should be used to configure the software instead of describing the hardware.

> you wouldn't need to alter intc_plic.c every time new vendor supports this, etc.

I actually prefer vendors to upstream their stuff, rather than keeping the configurations downstream in some repo somewhere.